rdrf/rdrf/views/form_view.py

```diff
@@ -885,6 +885,18 @@ def _get_formlinks(self, user, context_model=None):
         else:
             return []
 
+    def _set_initial_randomised_data(self, form_class, initial_data):
+        randomised_values = {
+            field_name: field.randomised_value for field_name, field in form_class.base_fields.items()
+            if hasattr(field, 'randomised_value')
+        }
+        if not randomised_values:
+            return initial_data
+        if not initial_data or initial_data == ['']:
+            initial_data = {}
```

Comment:
This part I still don't understand.
Without looking deeper into the code I see that `initial_data` can sometimes be a list  ( I guess when you have multi sections), but how it is alright then to always convert the initial_data to a dict.
So, I see a list is coming in and expect other code to be there that expects a list, but we modify it to a dict.
Wouldn't that cause problems for the code that expects a list?

Reply:
Updated the code to address these issues.